src/helpers/counterBuilder.js

```diff
@@ -0,0 +1,229 @@
+    var normalizedMaxDigits = {}
+
+    for (let period of this.state.periods) {
+      const minSegmentSize = this.getMinSegmentSize(period);
+      [normalizedMinDigits[period], normalizedMaxDigits[period]] = this.normalizeDigitLimitsForPeriod(
+        minDigits[period], maxDigits[period], minSegmentSize
+      )
+    }
+    return {
+      minDigits: normalizedMinDigits,
+      maxDigits: normalizedMaxDigits
+    }
+  }
+
+  normalizeDigitLimitsForPeriod (min, max, minSegmentSize) {
+    const minProvided = min !== undefined
+    const maxProvided = max !== undefined
+    function throwError () {
+      throw new Error(`conflict: minDigits (${min}) > maxDigits (${max})`)
+    }
+
+    if (minProvided && max) {
+      return (min > max) ? throwError() : [min, max]
+    } else if (minProvided && !maxProvided) {
+      return [min, (min > minSegmentSize) ? min : minSegmentSize]
+    } else if (!minProvided && max) {
+      return [(minSegmentSize > max) ? max : minSegmentSize, max]
+    } else {
+      return [minSegmentSize, maxProvided ? max : minSegmentSize]
+    }
+  }
+
+  /**
+   * Calculates maximum value for given period.
+   * @param {string} period
+   * @return {number}
+   */
+  getPeriodMaxValue (period) {
+    switch (period) {
+      case 'seconds':
+      case 'minutes':
+        return 59
+      case 'hours':
+        return 23
+      default:
+        if (period === this.props.minPeriod) {
+          // max number possible for this period during the countdown
+          return NumberCalculator.getPeriodNumberAt(
+            period, this.props.maxPeriod, this.state.initialTimeDiff
+          )
+        } else {
+          return this.getPeriodMaxValue(this.props.minPeriod)
+        }
+    }
+  }
+
+  /**
+   * Calculates initial numbers for all periods.
+   */
+  buildInitialNumbers () {
+    const timestamp = (this.props.direction === 'down') ? this.state.initialTimeDiff : 0
+    this.state = {
+      ...this.state,
+      numbers: NumberCalculator.calculateNumbers(this.state.periods, timestamp)
+    }
+  }
+
+  /**
+   * Validates counter's props.
+   * @param {Object} props - props to validate
+   */
+  static validateProps (props) {
+    if (props.seconds !== undefined) {
+      if (props.to !== undefined || props.from !== undefined) {
+        throw new Error('cannot use "to" and "from" with "seconds"')
+      } else if (props.seconds < 0) {
+        throw new Error('"seconds" must be greater than or equal to zero')
+      }
+    } else if (props.to === undefined) {
+      throw new Error('provide either "seconds" or "to"')
+    } else if (props.to < props.from) {
+      throw new Error('"to" must be bigger than "from"')
+    }
+    if (props.minDigits !== undefined && props.minDigits < 1) {
+      throw new Error('"minDigits" must be positive')
+    }
+    if (props.minPeriod && PERIODS.indexOf(props.minPeriod) < 0) {
+      throw new Error('"minPeriod" must be one of: days, hours, minutes, seconds')
```

Comment:
No idea, it is not critical, but if they will (e.g. adding milliseconds for some reasons) no test will fail